### PR TITLE
23 anforderungen an das gesamtprojekt aus dem leaderboard

### PR DESCRIPTION
**Pull request by JoKircher**
# Description

This pull request adds the requirements extracted from the Carla leaderboard web site as well as the use cases described there

Fixes # 23

## Time invested

Josef Kircher: 9 h
Simon Erlbacher: 2 h

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

## Does this PR introduce a breaking change?

no

## Most important changes

among the three main markdown files, the 08_use_cases.md is the largest and contains the most work

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (might be obsolete with CI later on)
- [x] New and existing unit tests pass locally with my changes (might be obsolete with CI later on)

### CODE REVIEW

File: doc/03_research/04_requirements/02_informations_from_leaderboard.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+* RoadOption.**CHANGELANELEFT**: Move one lane to the left.
+* RoadOption.**CHANGELANERIGHT**: Move one lane to the right.
+* RoadOption.**LANEFOLLOW**: Continue in the current lane.
+* RoadOption.**LEFT**: Turn left at the intersection.
+* RoadOption.**RIGHT**: Turn right at the intersection.
+* RoadOption.**STRAIGHT**: Keep straight at the intersection.
+
+**Important:** If the semantics of left and right are ambiguous, the next position should be used to clarify the path.
+
+**Important**: You are not allowed to make use of any privilege information offered by the CARLA simulator, including planners or any type of ground truth. Submissions using these features will be rejected and teams will be banned from the platform.
```
Comment (MarcoRiedenauer) [inline doc/03_research/04_requirements/02_informations_from_leaderboard.md:112]: Does this also count for training data? Or is it possible to train our nets e.g. with the output of the semantic camera?

File: doc/03_research/04_requirements/02_informations_from_leaderboard.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+* `Route completion:` Percentage of the route distance completed by an agent
+* `Infraction penalty:` The leaderboard tracks several types of infractions which reduce the score
+
+Every agent starts with a base infraction score of 1.0 at the beginning.
+
+Complete score is arithmetic mean of all routes driving scores.
+
+### Driving Score for route i
+
+Product of route completion of route i and Infraction penalty of route i
```
Comment (MarcoRiedenauer) [inline doc/03_research/04_requirements/02_informations_from_leaderboard.md:144]:
```suggestion
### Driving score for single route

Product of route completion and infraction penalty of this route
```

Sounds a bit better to me than the mathematical "i" :)

File: doc/03_research/04_requirements/03_requirements.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+## Prioritized driving aspects
+
+There are different ways to prioritize the driving aspects mentioned in the document [08_use_cases](https://github.com/ll7/paf22/blob/482c1f5a201b52276d7b77cf402009bd99c93317/doc/03_research/08_use_cases.md).
+The most important topics, in relation to this project, are the driving score and the safety aspect.
+Also, it is appropriate to implement the basic features of an autonomous car first. The list is a mixture of the different approaches. Prioritizing from very important functionalities to less important features.
+
+`Very important:`
+
+* Recognize the street limitations
```
Comment (MarcoRiedenauer) [inline doc/03_research/04_requirements/03_requirements.md:48]: Maybe "hold line" is also a very important task. Or is it included in "steering, accelerate, decelerate"?

**Review by MarcoRiedenauer — CHANGES_REQUESTED** [review]
A really well researched and prepared PR. All important traffic cases with corresponding informations and necessary actions are covered and the leaderboard rules are summarized.

I just have some minor improvements and questions :)

File: doc/03_research/04_requirements/02_informations_from_leaderboard.md (hunk 1 above)
Comment (JoKircher) [reply doc/03_research/04_requirements/02_informations_from_leaderboard.md:112]: Good question. I'm not quite sure, but certainly it should not be included in the submission as only RGB cameras are allowed. It might falsify our model if we train on data later unavailable, what do you think?

Comment (MarcoRiedenauer) [reply doc/03_research/04_requirements/02_informations_from_leaderboard.md:112]: Yes that’s right… So more work for us to create the labels ourselves, but at least we are then on the safe side 😄

**Review by MarcoRiedenauer — APPROVED** [review]

File: doc/03_research/04_requirements/02_informations_from_leaderboard.md (hunk 1 above)
Comment (JoKircher) [reply doc/03_research/04_requirements/02_informations_from_leaderboard.md:112]: Possibly there are training sets from the previous years

**Comment by ll7** [comment]
Very nice.

File: doc/03_research/04_requirements/02_informations_from_leaderboard.md (hunk 1 above)
Comment (JoKircher) [reply doc/03_research/04_requirements/02_informations_from_leaderboard.md:112]: @MarcoRiedenauer to follow up on the issue. In my opinion, for training data generation/annotated images, it should be possible to work with everything we have, for the submission we are limited to RGB cameras
